### DISCUSSION TITLE
old example not working to get raw i2s data into buffer array

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#731] [General]
https://github.com/pschatzmann/arduino-audio-tools/blob/c7aa6f915f6596fc36b86e231913c825c698b636/examples/i2s-serial/i2s-serial.ino

I wanted to get raw data into buffer and  in the process i test this but its not working .

**Comment by pschatzmann** [comment]
This examples does not exist any more since a long time. 
What old version are you still using ?
Please use the new API!

**Comment by aqildad-create** [comment]
how to get the raw data into buffer using new API ?

**Comment by pschatzmann** [comment]
Like in any other Arduino Stream with readBytes(). Just look at the class documentation of the I2SStream!
I created this documentation to avoid these kind of annoying questions!

In the Readme you can find links to the most important classes.
If you scroll down to Documentation you can find links to all classes alphabetically or by topic

**Reply by aqildad-create** [reply]
Yes apology but see i am using readbyte with AnalogAudioStream giving errors .

![image](https://user-images.githubusercontent.com/65456998/228788690-b4b6c9c6-5177-44c7-baab-d6eea93f9cb1.png)

**Comment by pschatzmann** [comment]
But you did not specify the buffer to copy the data to. Please learn how to use Arduino streams by reading how to read data e.g.
- from a file
- from Serial
